aea/helpers/search/models.py

```diff
@@ -651,6 +994,28 @@ def __eq__(self, other):  # pragma: nocover
         """Compare with another object."""
         return isinstance(other, And) and self.constraints == other.constraints
 
+    def encode(self) -> models_pb2.Query.ConstraintExpr.And():
```

Comment:
If you make it a class method you can do `cls._encode(...)`

Reply:
I don't understand this comment. 

Do you mean, changing this `encode` method to be a class method (and by extension all other `encode` methods)? 
If so, what's the advantage of doing that? This method **always** needs an instance of this class to encode, so if we change it to class method, the instance has to always be passed as a parameter. With an instance method, we can just call `encode` off of the instance itself.

Reply:
Sorry, I got confused here.